### PR TITLE
refactor(frontend): integrate new rovince territory state service implementation

### PR DESCRIPTION
**Pull request by sebastien-comeau**
### Description

As per title.

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/schemas/personal-informaton-service-schemas.server.ts [hunk 1]
```diff
@@ -247,7 +247,7 @@ export const personalInformationSchema = z.object({
       streetName: z.string().optional(),
       apartment: z.string().optional(),
       cityName: z.string().optional(),
-      provinceTerritoryStateId: z.string().optional(),
+      id: z.string().optional(),
```
Comment (nicholas-ly) [inline frontend/app/schemas/personal-informaton-service-schemas.server.ts:250]: I don't think this field should be renamed - `provinceTerritoryStateId` semantically differs from `id` of a mailing address.

(Same with line 240 above for home address)

File: frontend/app/mappers/personal-information-service-mappers.server.ts [hunk 2]
```diff
@@ -99,10 +99,10 @@ interface ToUpdateAddressRequestArgs {
   countryId?: string;
   streetName?: string;
   postalCode?: string;
-  provinceTerritoryStateId?: string;
+  id?: string;
```
Comment (nicholas-ly) [inline frontend/app/mappers/personal-information-service-mappers.server.ts:102]: Same comment about renaming the field in `personal-informaton-service-schemas.server`.

File: frontend/app/routes/public/apply/$id/child/review-adult-information.tsx [hunk 3]
```diff
@@ -63,12 +62,10 @@ export async function loader({ context: { configProvider, serviceProvider, sessi
...
   const t = await getFixedT(request, handle.i18nNamespaces);
   const locale = getLocale(request);
-  const lookupService = getLookupService();
 
   // Getting province by Id
-  const allRegions = localizeRegions(lookupService.getAllRegions(), locale);
-  const provinceMailing = allRegions.find((region) => region.provinceTerritoryStateId === state.contactInformation.mailingProvince);
-  const provinceHome = allRegions.find((region) => region.provinceTerritoryStateId === state.contactInformation.homeProvince);
+  const provinceMailingProvinceTerritoryStateAbbr = state.contactInformation.mailingProvince ? serviceProvider.getProvinceTerritoryStateService().findById(state.contactInformation.mailingProvince)?.abbr : undefined;
+  const provinceHomeProvinceTerritoryStateAbbr = state.contactInformation.homeProvince ? serviceProvider.getProvinceTerritoryStateService().findById(state.contactInformation.homeProvince)?.abbr : undefined;
```
Comment (nicholas-ly) [inline frontend/app/routes/public/apply/$id/child/review-adult-information.tsx:68]: Might be a nitpick but I feel "province" doesn't need to be in the variable name twice.

```suggestion
  const mailingProvinceTerritoryStateAbbr = state.contactInformation.mailingProvince ? serviceProvider.getProvinceTerritoryStateService().findById(state.contactInformation.mailingProvince)?.abbr : undefined;
  const homeProvinceTerritoryStateAbbr = state.contactInformation.homeProvince ? serviceProvider.getProvinceTerritoryStateService().findById(state.contactInformation.homeProvince)?.abbr : undefined;
```

This local variable naming applies to other routes in this PR

**Review by nicholas-ly — COMMENTED** [review]
I believe @Fbasham is working on removing the `/protected/personal-information` routes (and dependencies) so comments I made to those impacted files can be ignored.

**Comment by sebastien-comeau** [comment]
> I believe @Fbasham is working on removing the `/protected/personal-information` routes (and dependencies) so comments I made to those impacted files can be ignored.

Ok, I will wait for @Fbasham PR and I will rebase my branch over `main` in this case.

File: frontend/app/routes/public/apply/$id/child/review-adult-information.tsx (hunk 3 above)
Comment (sebastien-comeau) [reply frontend/app/routes/public/apply/$id/child/review-adult-information.tsx]: Fixed

File: frontend/app/mappers/personal-information-service-mappers.server.ts (hunk 2 above)
Comment (sebastien-comeau) [reply frontend/app/mappers/personal-information-service-mappers.server.ts]: Fixed, my bad.

File: frontend/app/schemas/personal-informaton-service-schemas.server.ts (hunk 1 above)
Comment (sebastien-comeau) [reply frontend/app/schemas/personal-informaton-service-schemas.server.ts]: Fixed, my bad.

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**